tools/hive-bigquery/hive_to_bigquery.py

```
@@ -0,0 +1,257 @@
+
+    try:
+        # Validates the user provided resources
+        logger.debug("Validating the resources")
+        if ResourceValidator.validate(hive_component, gcs_component,
+                                      bq_component):
+            logger.info("All the provided resources are valid")
+        else:
+            logger.error("Check the provided resources")
+            print_and_log("Check the log file for detailed errors",
+                          logging.CRITICAL)
+            exit()
+
+        hive_table_object = HiveTable(
+            hive_component, PropertiesReader.get('hive_database'),
+            PropertiesReader.get('hive_table_name'),
+            PropertiesReader.get('incremental_col')
+        )
+        # Wrapper to describe Hive table resource
+        hive_table_model = hive_table_object.hive_table_model
+        logger.debug(hive_table_model)
+
+        bq_table_object = BigQueryTable(PropertiesReader.get('dataset_id'),
+                                        PropertiesReader.get('bq_table'),
+                                        hive_table_model)
+        # Wrapper to describe BigQuery table resource
+        bq_table_model = bq_table_object.bq_table_model
+        logger.debug(bq_table_model)
+
+    except Exception as error:
+        logger.exception(error)
+        print_and_log("Check the log file for detailed errors",
+                      logging.CRITICAL)
+        exit()
+
+    try:
+        # Verifies whether the tracking table exists from the previous run
+        mysql_component.verify_tracking_table(hive_table_model)
+        # Validates the bq_table_write_mode provided by the user
+        if not bq_component.check_bq_write_mode(mysql_component,
+                                                hive_table_model,
+                                                bq_table_model):
+            print_and_log("Check the log file for detailed errors",
+                          logging.CRITICAL)
+            exit()

```

Should almost never use exit() You should raise an error here, see above.